lib_eio_posix/eio_posix_stubs.c

```diff
@@ -0,0 +1,107 @@
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <sys/eventfd.h>
+#include <sys/random.h>
+#include <sys/syscall.h>
+#include <limits.h>
+#include <errno.h>
+#include <dirent.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/uio.h>
+
+#include <caml/mlvalues.h>
+#include <caml/memory.h>
+#include <caml/alloc.h>
+#include <caml/signals.h>
+#include <caml/unixsupport.h>
+#include <caml/bigarray.h>
+
+#ifdef ARCH_SIXTYFOUR
+#define Int63_val(v) Long_val(v)
+#else
+#define Int63_val(v) (Int64_val(v)) >> 1
+#endif
+
+CAMLprim value caml_eio_posix_getrandom(value v_ba, value v_off, value v_len) {
+  CAMLparam1(v_ba);
+  ssize_t ret;
+  ssize_t off = (ssize_t)Long_val(v_off);
+  ssize_t len = (ssize_t)Long_val(v_len);
+  do {
+    void *buf = Caml_ba_data_val(v_ba) + off;
+    caml_enter_blocking_section();
+    ret = getrandom(buf, len, 0);
```

Comment:
`getrandom` is not available on macOS it would seem, there's `getentropy` and `arc4random` which look like they may be a good fit?

Reply:
ack, getentropy() would be the thing. in fact getrandom() was designed after getentropy